content/lessons/chapter-3/pool-1.tsx

```diff
@@ -1,105 +1,186 @@
+      setFinalMining(false)
+    }
+  }, [blocks1, blocks2])
+
+  const turnOnButton = async () => {
+    if (step === 0) {
+      setShowText(false)
+      await sleep(100)
+      setStep(1)
+      setFinalMining(true)
+    }
+  }
 
   return (
-    <div className="flex flex-col items-center gap-[30px] px-[15px] py-[30px] md:px-[75px] md:py-[75px]">
-      <div className="flex h-full max-h-[69px] w-full max-w-[800px] flex-col items-start gap-[10px] text-white">
-        <div className="flex items-center justify-between gap-[10px] self-stretch py-[2px]">
-          <span className="h-[25px] text-left font-nunito text-[18px] font-semibold text-white">
-            <span>
-              {PROTAGONISTS.reduce((acc, profile) => acc + profile.value, 0)}
-            </span>
-            <span> blocks</span>
-          </span>
-          <span className="h-[25px] text-right font-nunito text-[18px] font-semibold text-white">
-            <span>
-              {ANTAGONISTS.reduce((acc, profile) => acc + profile.value, 0)}
-            </span>
-            <span> blocks</span>
-          </span>
-        </div>
-        <ContributionBar
-          total={TOTAL_BLOCKS}
-          protagonists={PROTAGONISTS}
-          antagonists={ANTAGONISTS}
-        />
-      </div>
-      <div className="flex gap-[30px]">
-        {PROFILES.map((profile, i) => (
+    <div className="flex flex-col items-center gap-[30px] py-[30px] px-[15px] md:py-[75px] md:px-[75px]">
+      <HashrateChallenge
+        setHashPower={setHashPower}
+        totalBlocks={TOTAL_BLOCKS}
+        turnOnButton={turnOnButton}
+        step={step}
+        hashPower={hashPower}
+        protagonists={PROTAGONISTS}
+        antagonists={ANTAGONISTS}
+        profiles={PROFILES.map((profile, i) => (
           <Profile
             key={i}
             username={profile.username}
             avatar={profile.avatar}
             description={profile.description}
           >
             <Card className="flex">
-              <span className="fade-in font-nunito text-[15px] font-bold text-white text-opacity-25">
+              <span
+                className={clsx('fade-in font-nunito text-[15px] font-bold', {
+                  'text-white text-opacity-25': step === 0,
+                  'fade-in text-[#EDA081]': step !== 0,
+                })}
+              >
                 Blocks found
               </span>
-              <span className="fade-in font-nunito text-[15px] font-bold text-white text-opacity-25">
-                0
+              <span
+                className={clsx(
+                  'fade-in font-space-mono text-[15px] font-normal text-white',
+                  {
+                    'text-opacity-25': step === 0,
+                  }
+                )}
+              >
+                {i === 2 || i === 4
+                  ? i === 2
+                    ? PROTAGONISTS[2].value
+                    : ANTAGONISTS[0].value
+                  : 0}
```

Comment:
This is a bit confusing, I assume you have to do this index checking because `<HashrateChallenge>` takes a profiles prop, I think you can maybe get around having to do these conditionals if instead you make it take a protagonists and antagonists array instead.